crates/tarball/src/lib.rs

```diff
@@ -122,9 +121,12 @@ pub struct DownloadTarballToStore<'a> {
 }
 
 impl<'a> DownloadTarballToStore<'a> {
-    /// Execute the subroutine.
-    pub async fn run(self) -> Result<Arc<HashMap<OsString, PathBuf>>, TarballError> {
-        let &DownloadTarballToStore { tarball_cache, package_url, .. } = &self;
+    /// Execute the subroutine with cache.
+    pub async fn with_cache(
```

Comment:
Nit:

```suggestion
    pub async fn run_with_cache(
```

Reply:
> a function should be a verb

BTW, Rust naming convention isn't like that.